stl/src/syncstream.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// initialize syncstream mutex map
+
+#include <internal_shared.h>
+#include <map>
+#include <shared_mutex>
+
+#pragma warning(disable : 4074)
+#pragma init_seg(compiler)
+
+_STD_BEGIN
+// OBJECT DECLARATIONS
+struct _Mutex_count_pair {
+    shared_mutex _Mutex;
+    uint64_t _Ref_count = 0;
+};
+
+template <class _Ty>
+class _Crt_allocator {
+public:
+    using value_type                             = _Ty;
+    using size_type                              = size_t;
+    using difference_type                        = ptrdiff_t;
+    using propagate_on_container_move_assignment = true_type;
+    using is_always_equal                        = true_type;
+
+    constexpr _Crt_allocator() noexcept = default;
+
+    constexpr _Crt_allocator(const _Crt_allocator&) noexcept = default;
+    template <class _Other>
+    constexpr _Crt_allocator(const _Crt_allocator<_Other>&) noexcept {}
+
+    _NODISCARD __declspec(allocator) _Ty* allocate(_CRT_GUARDOVERFLOW const size_t _Count) {
+        auto _Ptr = _calloc_crt(_Count, sizeof(_Ty));
+        if (!_Ptr) {
+            throw bad_alloc{};
+        }
+        return static_cast<_Ty*>(_Ptr);
+    }
+
+    void deallocate(_Ty* const _Ptr, const size_t) noexcept {

```

Comment:
We conventionally don't mark unnamed parameters as `const` (no danger of modifying them :smile_cat:):
```suggestion
    void deallocate(_Ty* const _Ptr, size_t) noexcept {
```